### PR TITLE
Final API cleanup scheduler implementation

### LINKED ISSUE #8197: Final api cleanup
- [x] Drop any api that used only by the old agent on the server 
- [x] Start stripping the underlying code , ticketize if stuck  
- [x] resource_Action_update, deploy_Start and deploy_done are to be investigated with care 
- [x] check the config option and env settings

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

This PR removes API endpoints, used between the server and the agent, that are not used anymore in the new scheduler implementation.

Part one #8197 

# Self Check:

- [x] Attached issue to pull request
- [ ] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [ ] ~~End user documentation is included or an issue is created for end-user documentation~~
- [ ] ~~If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)~~

### CODE REVIEW

**Comment by arnaudsjs** [comment]
I still need to add:

* A database migration script that removes the old environment settings.
* Update the documentation pages.

I will add both changes to a separate PR to get this merged asap.

**Comment by wouterdb** [comment]
Dang, cool

File: src/inmanta/protocol/methods.py [hunk 1]
```diff
@@ -441,23 +441,6 @@ def get_resource(
...
-    :param sid: Optional. Session id of the agent (transparently added by agent client).
-    :param version: Optional. The version to retrieve. If none, the latest available version is returned. With a specific
-                    version that version is returned, even if it has not been released yet.
-    :param incremental_deploy: Optional. Indicates whether the server should only return the resources that changed since the
-                               previous deployment.
-    """
-
-
 @method(path="/resource", operation="POST", agent_server=True, arg_options=ENV_OPTS, client_types=[const.ClientType.agent])
 def resource_action_update(
```
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply src/inmanta/protocol/methods.py:445]: I think so. It's still used by the getfact and dryrun functionality.

[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/protocol/methods.py:445]: I wonder if we could move it to scheduler side, could you make a ticket for that? 
(i.e. we can't move the other parts of dryrun and get_fact, but perhaps we can move this bit?)

**Review by wouterdb — APPROVED** [review]

File: changelogs/unreleased/8197-final-api-cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+description: Cleanup API endpoint and configuration options are are no longer used.
```
Comment (Hugo-Inmanta) [inline changelogs/unreleased/8197-final-api-cleanup.yml:2]:
```suggestion
description: Clean up API endpoints and configuration options that are no longer used.
```

File: docs/install/3-configure-agents.rst [hunk 3]
```diff
@@ -143,7 +143,6 @@ The manually-started agent can be configured via a ``/etc/inmanta/inmanta.d/*.cf
 configure the behavior of the manually started agent:
 
 * :inmanta.config:option:`config.state-dir`
-* :inmanta.config:option:`config.agent-names`
 * :inmanta.config:option:`config.environment`
 * :inmanta.config:option:`config.agent-map`
```
Comment (Hugo-Inmanta) [inline docs/install/3-configure-agents.rst:147]:
```suggestion

```

File: changelogs/unreleased/8197-final-api-cleanup.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      - server.auto-recompile-wait
+      - config.agent-interval
+      - config.agent-splay
+      - server_rest_transport.port
+    The following environment settings were removed:
+      - push_on_auto_deploy
+      - agent_trigger_method_on_auto_deploy
+      - environment_agent_trigger_method
+      - autostart_agent_map
+    The following internal API endpoint were removed:
```
Comment (Hugo-Inmanta) [inline changelogs/unreleased/8197-final-api-cleanup.yml:24]:
```suggestion
    The following internal API endpoints were removed:
```

File: tests/test_server.py [hunk 5]
```diff
@@ -713,10 +693,15 @@ async def test_resource_action_log(server, client, environment):
 
 async def test_invalid_sid(server, client, environment):
     """
-    Test the server to manage the updates on a model during agent deploy
+    Verify that API endpoints, that should only be called by an agent, return an HTTP 400
+    if they are called without a session id.
     """
     # request get_code with a compiler client that does not have a sid
```
Comment (Hugo-Inmanta) [inline tests/test_server.py:699]:
```suggestion

```

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request closed without merging**